packages/discord.js/src/structures/interfaces/InteractionResponses.js

```diff
@@ -100,7 +101,7 @@ class InteractionResponses {
    */
   async reply(options) {
     if (this.deferred || this.replied) throw new DiscordjsError(ErrorCodes.InteractionAlreadyReplied);
-    this.ephemeral = options.ephemeral ?? false;
+    this.ephemeral = options.ephemeral ?? new MessageFlagsBitField(options.flags).has(MessageFlags.Ephemeral);
```

Comment:
mmmmh, this is already fully computed here:
https://github.com/discordjs/discord.js/blob/67360a9e0bae361d2465379cc754554a791a1b9c/packages/discord.js/src/structures/MessagePayload.js#L149-L164

I'd love to avoid the double instantiation of `MessageFlagsBitFields`, but that doesn't seem easy in this case. At most though, I feel like we should remove this line all-together (since `MessagePayload` checks for `options.ephemeral` anyways, see line 162) and instead just do `this.ephemeral = new MessageFlagsBitField(body.flags).has(MessageFlags.Ephemeral);` just a bit later to avoid duplicating any logic.